python/foglamp/services/common/service_announcer.py

```diff
@@ -6,52 +6,60 @@
 
 """Common FoglampMicroservice Class"""
 
-import foglamp.services.common.avahi as avahi
+
+import socket
+from zeroconf import ServiceInfo, Zeroconf
+
 from foglamp.common import logger
-import dbus
 
-__author__ = "Mark Riddoch"
+
+__author__ = "Mark Riddoch, Amarendra K Sinha"
 __copyright__ = "Copyright (c) 2017 OSIsoft, LLC"
 __license__ = "Apache 2.0"
 __version__ = "${VERSION}"
 
 _LOGGER = logger.setup(__name__)
 
-class ServiceAnnouncer:
-    _service_name = None
-    """ The name of the service to advertise """
 
-    _group = None
-    """ The Avahi group """
+class ServiceAnnouncer:
+    def __init__(self, sname, stype, port, txt):
+        host_name = socket.gethostname()
+        host = self.get_ip()
+        service_name = "{}_{}.{}".format(host_name, sname, stype)
```

Comment:
can we simpy keep it the name, as passed? hostname we are already putting in for `server`